optuna/trial.py

```diff
@@ -357,21 +411,32 @@ def report(self, value, step=None):
 
         Example:
 
-            Report intermediate scores of `SGDClassifier <https://scikit-learn.org/stable/modules/
-            generated/sklearn.linear_model.SGDClassifier.html>`_ training
```

Comment:
IMO, this description is useful for users because it clarifies the usage of the `report` function. What do you think of it?

Reply:
Thanks!
Yes, I think this document is useful for users.
But I'm not sure why #702 deleted the document on Example.
Is there any criteria to delete documents?

Reply:
I guess that is because the description in the previous commit is not valid. 

>  Get an objective value and a value of parameter ``x`` in the first row.

In the previous example, the code actually obtains the objective value and parameter value as follows:

```python
>>> df.value[0]	
        0.0
>>> df.params.x[0]
        1.0
```

But they are replaced by the following assertions in the new example:

```python
assert isinstance(df, pandas.DataFrame)
assert df.shape[0] == 3  # n_trials.
```

> Is there any criteria to delete documents?

I don't think we have such criteria because we just started using the doctest.

@hvy @sile @g-votte 

Do you have any comments about the description (or title?) of the doctest code?